src/server.c

```diff
@@ -1168,17 +1168,26 @@ void cronUpdateMemoryStats() {
 
 /* Pause clients on write as long as OOM and pending lazy-free jobs in background.
  * By doing so the server might avoid unnecessary eviction or rejection. */
-static void ClientWritePauseDuringOOM() {
-    
+static void clientWritePauseDuringOOM() {
     /* if OOM and available pending lazyfree jobs then pause client-write to avoid overloading. 
      * Pause also evictions, since we might reach below maxmemory after lazyfree */
     if ((getMaxmemoryState(NULL,NULL,NULL,NULL) == C_ERR) &&
         (bioPendingJobsOfType(BIO_LAZY_FREE))) {
-        pauseActions(PAUSE_OOM_THROTTLE,LLONG_MAX, 
-                     PAUSE_ACTION_EVICT | PAUSE_ACTION_CLIENT_WRITE );
+        if (!server.current_client_paused_oom_time) {
```

Comment:
i'm not a fan of the new `init_flow`, and even if we proceed with it, i suppose we need to patch a few other stat variables.
correct me if i'm wrong, but it seems to me that this RESETSTAT didn't have a real negative impact on your code.
i.e. it would zero current_client_paused_oom_time, causing us to start a new measurement next time we get here, so it doesn't have any negative impact on the stats.

the unexpected impact is that we'll call `pauseActions` again (even though we're already paused), but that also doesn't have any negative impact, right?
ohh, the risk is that if right after RESETSTAT we realize we need to drop the pause, we won't call unpauseActions.
maybe the right solution is to separate between the decision on when we should call pauseActions and unpauseActions from any code related to the metrics?

Reply:
IMO, just because RESETSTAT can be  risky and can be called from two, very different flows, I feel it is the right thing to reflect it in its function declaration with `init_flow` flag rather than hide it and have "clean and optimistic picture" of it. Adding another argument might complicate a rather simple logic.

Reply:
i think resetServerStats is doing exactly what it should do (clear a bunch of variable).
i might not mind the extra argument, but i don't like the complicated logic inside that function.
the problem here is that we bind other logic (calling `unpauseActions`) to a stat variable.
i think that's the root of this issue.